src/external_cli/arg_builder.rs

```diff
@@ -0,0 +1,122 @@
+/// A helper to make passing arguments to [`Command`](std::process::Command) more convenient.
+#[derive(Debug, Clone)]
+pub struct ArgBuilder(Vec<String>);
+
+impl ArgBuilder {
+    /// Create a new builder for command arguments.
+    pub fn new() -> Self {
```

Comment:
```suggestion
    pub const fn new() -> Self {
```

This could be made constant, as long as there are no plans to add more complexity to this function in the future.